### ISSUE TITLE
Media source handling in file tree

### ISSUE THREAD

**Issue by jcurtis** [modxbot/migrate#7565]
jcurtis created Redmine issue ID 7565

I've setup a log in for a client where they have access to the file tree, but I've removed access to media source with ID 1 - the whole file tree.

When they log in and look at their file tree, the default selection in the drop box is '1' with only a folder titled 'Media' below it.

When expanded, only the media sources they have permissions to view are available and from that point on all is OK.

Should the media source be checked for permission and the first available one loaded rather than assuming one can be loaded but then failing to initialise it due to permissions?

**Comment by MarkH** [comment]
markh submitted:

Did you check this in 2.2.1 or 2.2.0? I'd swear I saw a fix for that coming by for 2.2.1...

**Comment by jcurtis** [comment]
jcurtis submitted:

Definitely in 2.2.1-pl

**Comment by brianlarson** [comment]
brianlarson submitted:

I discovered the same problem just now. I created a few new Media Sources and saw that it defaulted to the one with an ID of 1 (it's like new one(s) pushed the Filesystem default became an ID of 2—or maybe I screwed that up). I had to go in and manually delete all the Media Sources I created in phpMyAdmin and then change the ID of Filesystem back to 1. That restored all back to normal.

**Comment by gadamiak** [comment]
gadamiak submitted:

There's a @default_media_source@ setting for handling that.

**Comment by chrisfullman** [comment]
chrisfullman submitted:

Is there any update on this? I'm experiencing the same issue on 2.2.9. I created a new Media Source (called "Site Content") that has the base path set to a subfolder (content/) in the original Filesystem, and have no specific permissions set to the new Media Source, making that available for any user to manage and upload to.

The admin account can view the entire folder tree and upload content to it. The publisher user only sees "Site Content" (occasionally appearing as a 2) and "Media" in the folder tree.

This is a relatively clean install and I'm just getting started with setting up user permissions. I verified that the Publisher permission set has full access to files and media sources.
